pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -462,6 +468,19 @@ func SetupIndexes(ctx context.Context, indexer client.FieldIndexer) error {
 	return jobframework.SetupWorkloadOwnerIndex(ctx, indexer, gvk)
 }
 
+func CanSupportIntegration(log logr.Logger, opts ...jobframework.Option) bool {
+	options := jobframework.DefaultOptions
+	for _, opt := range opts {
+		opt(&options)
+	}
+	v := options.KubeServerVersion.GetServerVersion()
+	if v.String() == "" || v.LessThan(kubeversion.KubeVersion1_27) {
+		log.Error(errPodNoSupportKubeVersion, "Failed to configure reconcilers", "kubernetesVersion", v)
+		os.Exit(1)
```

Comment:
A bit weird that this function can return true/false, but we are exiting instead.

Maybe return an error and log the error in the caller?

Reply:
I implemented this as currently viewed since other integrations (not pod) are possible to customize whether the `a. manager exits with 1` or `b. just skips to register the integration to the manager`.

However, I'm ok with enforcing `a` or `b`. @alculquicondor WDYT?

Reply:
I see what you mean. I think we can enforce 1 for now, as Pod is the only one.

In general, I think it's more useful to fail startup than silently skip.

Reply:
It makes sense.

Reply:
Done.